File: javascript/sieve/sieve.js
export const primes = (max) => {
    const sieve = [];
    const result = [];

    for (let i = 2; i <= max; i++) {
        if (!sieve[i]) {
            result.push(i);
        }

        for (let j = i; j <= max; j += i) {
            sieve[j] = true;
        }
    }

    return result;
};

File: clojure/sieve/src/sieve.clj
(ns sieve)

(defn sieve [max]
  (:primes
   (reduce (fn [{:keys [primes marked]} n]
             {:primes (if (marked n) primes (conj primes n))
              :marked (reduce conj marked (range (* 2 n) (inc max) n))})
           {:primes [] :marked #{}}
           (range 2 (inc max)))))
